packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs

```diff
@@ -147,7 +172,7 @@ internal IReadOnlyList<ParameterProvider> GetUriParameters()
 
         protected override FieldProvider[] BuildFields()
         {
-            List<FieldProvider> fields = [EndpointField];
+            HashSet<FieldProvider> fields = [EndpointField];
```

Comment:
I think we should not make this a HashSet.
What is the purpose of making this a set?
A set might make the order to change unexpectedly. We would not want to see that.

Reply:
I thought the list of fields should be unique so making it a set would ensure that. If the order changes though that may not be desirable